scripts/msct_image.py

```diff
@@ -1048,6 +1009,152 @@ def change_data_orientation(data, old_orientation='RPI', orientation="RPI"):
+                    dtype = np.int8
+                elif max_in <= np.iinfo(np.int16).max and min_in >= np.iinfo(np.int16).min:
+                    dtype = np.int16
+                elif max_in <= np.iinfo(np.int32).max and min_in >= np.iinfo(np.int32).min:
+                    dtype = np.int32
+                elif max_in <= np.iinfo(np.int64).max and min_in >= np.iinfo(np.int64).min:
+                    dtype = np.int64
+                else:
+                    raise ValueError("Maximum value of the image is to big to be represented.")
+        else:
+            # if max_in <= np.finfo(np.float16).max and min_in >= np.finfo(np.float16).min:
+            #    type = 'np.float16' # not supported by nibabel
+            if max_in <= np.finfo(np.float32).max and min_in >= np.finfo(np.float32).min:
+                dtype = np.float32
+            elif max_in <= np.finfo(np.float64).max and min_in >= np.finfo(np.float64).min:
+                dtype = np.float64
+
+        dtype = to_dtype(dtype)
+    else:
+        dtype = to_dtype(dtype)
+
+        # if output type is int, check if it needs intensity rescaling
+        if "int" in dtype.name:
+            # get min/max from output type
+            min_out = np.iinfo(dtype).min
+            max_out = np.iinfo(dtype).max
+            # before rescaling, check if there would be an intensity overflow
+
+            if (min_in < min_out) or (max_in > max_out):
+                # This condition is important for binary images since we do not want to scale them
+                sct.printv('WARNING: To avoid intensity overflow due to convertion to '+dtype.name+', intensity will be rescaled to the maximum quantization scale.', 1, 'warning')
+                # rescale intensity
+                data_rescaled = src.data * (max_out - min_out) / (max_in - min_in)
+                dst.data = data_rescaled - ( data_rescaled.min() - min_out )
+
+    # change type of data in both numpy array and nifti header
+    dst.data = getattr(np, dtype.name)(dst.data)
+    dst.hdr.set_data_dtype(dtype)
+    return dst
+
+
+def to_dtype(dtype):
+    """
+    Take a dtypeification and return an np.dtype
+
+    :param dtype: dtypeification (string or np.dtype or None are supported for now)
+    :return: dtype or None
+    """
+    # TODO add more or filter on things supported by nibabel
+
+    if dtype is None:
+        return None
+    if isinstance(dtype, type):
+        try:
+            if isinstance(dtype(0).dtype, np.dtype):
+                 return dtype(0).dtype
+        except: # TODO
+            raise
+    if isinstance(dtype, np.dtype):
+        return dtype
+    if isinstance(dtype, str):
+        return np.dtype(dtype)
+
+    raise TypeError("data type {}: {} not understood".format(dtype.__class__, dtype))
+
+def zeros_like(img, dtype=None):
+    dst = change_type(img, dtype)
+    dst.data[:] = 0
+    return dst
+
+def empty_like(img, dtype=None):
```

Comment:
not sure i understand the origin of the name `empty_like`. I find it might get confused with `zeros_like`. It just does a copy of an Image object, right? so why not calling it `copy()`? Or maybe i am missing something

Reply:
The implementation of `empty_like()` is indeed performing a copy from now, but empty_like is a standard function of numpy which just allocates but doesn't initialize storage. Using this over zeros_like() or copy() makes the intent clearer.

Reply:
i see. so, what would be example cases where one would prefer to use empty_like() vs. copy()?

Reply:
I used it for things like:
```python
scripts/sct_compute_hausdorff_distance.py:            self.thinned_image = msct_image.empty_like(self.image)
scripts/sct_compute_hausdorff_distance.py-            self.thinned_image.data = self.zhang_suen(self.image.data)
scripts/sct_compute_hausdorff_distance.py-            self.thinned_image.absolutepath = sct.add_suffix(self.image.absolutepath, "_thinned")
```